File: python/lswf.py
fib = []
fib.append(1)
fib.append(1)

def fibonacci(N):
    for i in range(2, N):
        fib.append(fib[i - 1] + fib[i - 2])

        if fib[i] > N:
            break
            
    return i


with open('input.txt', 'r') as fin:
    N = int(fin.readline())

lst = fibonacci(N)

fib.reverse()
print()

somma = 0
seq = []
for i in range(0,len(fib)-1):
    potSomma = somma + fib[i]
    if potSomma < N:
        somma = potSomma
        seq.append(1)
    else:
        seq.append(0)

seq.append(1)
seq.reverse()

if N == 1 or N > 4:
    seq.pop()

with open('output.txt', 'w') as fout:
    for i in seq:
        fout.write(str(i))

File: ruby/lswf.rb
fib = []
fib.push 1
fib.push 1

def fibonacci(n, fib)
  for i in 2...n do
    fib.push fib[i-1] + fib[i-2]

    break if fib[i] > n
  end

  return i
end

File.open('input.txt', 'r') do |fin|
  N = fin.gets.to_i
end

lst = fibonacci(N, fib)

fib = fib.reverse

somma = 0
seq = []

for i in 0...fib.length-1 do
  potSomma = somma + fib[i]

  if potSomma < N
    somma = potSomma
    seq << 1
  else
    seq << 0
  end
end

seq << 1
seq = seq.reverse

seq.pop if N == 1 || N > 4

File.open('output.txt', 'w') do |fout|
  seq.each {|c| fout.print c}
end
